cloudinit/stages.py

```diff
@@ -248,17 +248,40 @@ def _get_data_source(self, existing):
             LOG.debug(myrep.description)
 
         if not ds:
-            util.del_file(self.paths.instance_link)
```

Comment:
In my testing on ec2 with this branch, the 'check-cache' always finds the instance obj.plk,  which AFAICT, means that we never go down the DataSourceNotFound.

The only time I could force this was if the obj.pkl did not exist, then find_source() would call update_metadata() which calls get_data().  If IMDS is down, then we return False and do not find any datsources.  This triggers the DataSourceNotFoundException path.

Down that path, it then calls self._restore_from_checked_cache() which will fail the same way (no obj.plk)  so, I cannot see how the rest of the exception handler is doing anything.  What am I missing?